asio/include/asio/yield.hpp

```diff
@@ -10,14 +10,27 @@
 
 #include "coroutine.hpp"
 
+#if defined(__clang__)
```

Comment:
Do we need something that's compiler-portable here rather than just the GCC pragma? Should we be using Hedley to handle the suppression? I'm not sure that this pragma will work on PS or Switch which use clang. 
Given that we include `yield.hpp` at the top and `unyield.hpp` at the bottom, I'm wondering if we should make our own wrappers for those files in the SDK itself which handle these warnings and can use Hedley to handle this more portably.

Reply:
Hedley dont contains anything to handle the warning suppresion for Implicit-fallthough, only `HEDLEY_FALL_THROUGH` but i dont work for macro.

I also tried to add the `push/diagnostic ignored wimplicit` in `yield.hpp` and pop in `unyield.hpp` , it will also supress the warnings who are not contained in the macro, do we really want that ?

what i will suggest, will be to modify the macro i created to make it more compiler-portable.

What do you think about @stephenwhittle  ?

Reply:
Yep, making it more compiler portable will be fine and then this will be approved